tools/check_tidy.py

```diff
@@ -14,113 +14,166 @@
+
+    def _check_clang_format(self, base):
+        clang_format = spawn.find_executable(base)
+
+        if not clang_format:
+            clang_format = spawn.find_executable("clang-format")
+            if clang_format:
+                print("%sUsing %s instead of %s%s"
+                      % (ex._TERM_YELLOW, clang_format, base, ex._TERM_EMPTY))
+            else:
+                print("%sNo %s found, skipping checks!%s"
+                      % (ex._TERM_RED, base, ex._TERM_EMPTY))
+
+        self._clang_format = clang_format
 
-def is_checked_by_clang(file):
+    @property
+    def error_count(self):
+        return len(self.diffs)
+
+    def is_checked_by_clang(self, file):
+        _, ext = fs.splitext(file)
+        return ext in self._extensions and file not in self._skip_files
+
+    def check(self, files):
+        if not self._clang_format:
+            return
+
+        for file in filter(self.is_checked_by_clang, files):
+            output = ex.run_cmd_output(self._clang_format,
+                                       ['-style=file', file],
+                                       quiet=True)
+
+            with tempfile.NamedTemporaryFile() as temp:
+                temp.write(output)
+                temp.flush() # just to be really safe
+                self._diff(file, temp.name)
+
+    def _diff(self, original, formatted):
+        try:
+            subprocess.check_output(['diff', '-u', original, formatted])
+        except subprocess.CalledProcessError as error:
+            # if there is a difference between the two files
+            # this error will be generated and we can extract
+            # the diff from that it. Otherwise nothing to do.
+            self.diffs.append(error.output.decode())
+
+
+def is_interesting(exts, skip_files, dir_path, file):
     _, ext = fs.splitext(file)
-    return ext in clang_format_exts and file not in skip_files
+    return ext in exts and file not in skip_files
 
 
 def check_tidy(src_dir):
-    count_lines = 0
-    count_empty_lines = 0
-
-    for (dirpath, dirnames, filenames) in os.walk(src_dir):
-        if any(d in fs.relpath(dirpath, src_dir) for d in skip_dirs):
-            continue
-
-        files = [fs.join(dirpath, name) for name in filenames
-                 if is_interesting(name)]
-
-        if not files:
-            continue
-
-        for file in files:
-            if is_checked_by_clang(file):
-                formatted = ex.run_cmd_output(' '.join(['clang-format-3.8',
-                                                        '-style=file', file]),
-                                              quiet=True)
-                f = open(file + '.formatted', 'w')
-                f.write(formatted)
-                f.close()
-                ex.check_run_cmd('diff', [file, file+'.formatted'], quiet=True)
-                fs.remove(file + '.formatted')
+    exts_allowed = ['.c', '.h', '.js', '.py', '.sh', '.cmake']
```

Comment:
That's a good question. I'm using the same extensions which were already in place. Maybe someone more knowledgeable could answer that question.